Update Go version in backporting Dockerfile

tbalthazar opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

Trying to run the `dev-doctor` script from the backporting image fails:
```
$ docker run -e GITHUB_TOKEN -v $(pwd):/cilium -v "$HOME/.ssh":/home/user/.ssh -it cilium-backport /bin/bash
user@9c66b4f87c78:/$ cd cilium/
user@9c66b4f87c78:/cilium$ go run ./tools/dev-doctor --backporting
build github.com/cilium/cilium/tools/dev-doctor: cannot load io/fs: malformed module path "io/fs": missing dot in first path element
```

The `golang` package on Ubuntu 20.04 installs Go 1.13. The `io/fs` package was added in Go 1.16:
https://tip.golang.org/doc/go1.16#fs

This installs the latest Go version.

Fixes: #21121

tbalthazar committed:
Update Go version in backporting Dockerfile

Trying to run the `dev-doctor` script from the backporting image fails:
```
$ docker run -e GITHUB_TOKEN -v $(pwd):/cilium -v "$HOME/.ssh":/home/user/.ssh -it cilium-backport /bin/bash
user@9c66b4f87c78:/$ cd cilium/
user@9c66b4f87c78:/cilium$ go run ./tools/dev-doctor --backporting
build github.com/cilium/cilium/tools/dev-doctor: cannot load io/fs: malformed module path "io/fs": missing dot in first path element
```

The `golang` package on Ubuntu 20.04 installs Go 1.13.
The `io/fs` package was added in Go 1.16:
https://tip.golang.org/doc/go1.16#fs

This installs the latest Go version.

Fixes: #21121

Signed-off-by: Thomas Balthazar <thomas@balthazar.info>

tklauser commented on contrib/backporting/Dockerfile:
```diff
@@ -1,17 +1,19 @@
 FROM ubuntu:20.04
 
 LABEL maintainer="maintainer@cilium.io"
+ARG GOLANG_VERSION=1.19.3
```
Will this get updated by `make update-go-version` as well? If not, it that `make` target should be extended to update it.

tbalthazar replied:
Good point @tklauser, I wasn't aware of that target in the Makefile.
I updated the PR!

tbalthazar committed:
Update 'update-go-version' target in Makefile

`contrib/backporting/Dockerfile` now installs the Go version currently
used by the project.

This updates the `update-go-version` target to take that file into
account.

Signed-off-by: Thomas Balthazar <thomas@balthazar.info>

tklauser reviewed (approved):
Thanks @tbalthazar 🚀
